### PR TITLE
Add System theme setting

### LINKED ISSUE #4057: Have a "Auto" setting for dark mode
Maybe make that the default, then "auto" can select light/dark with https://www.hanselman.com/blog/how-to-detect-if-the-users-os-prefers-dark-mode-and-change-your-site-with-css-and-js

Twitter thread: https://twitter.com/smurthys/status/1569782727624327171?s=20&t=CnmVxZacMIlXUPziwptCOg

### PR DESCRIPTION
**Pull request by RubenRBS**
With this new setting, we query the browser every time the settings change and set default/dark based on the `prefers-color-scheme: dark` response.
Note that this is different from the initial guessing we do to set the site to default/dark based on the same query, as that one does not get updated after being initialized.

My only question now is if this should be the new default, which I dont think is the case, but it's better to ask

Closes #4057

### CODE REVIEW

File: static/settings.ts [hunk 1]
```diff
@@ -413,9 +413,18 @@ export class Settings {
         enableAllSchemesCheckbox.on('change', this.onThemeChange.bind(this));
 
         $.data(themeSelect, 'last-theme', themeSelect.val() as string);
+        this.onThemeChange();
```
Comment (mattgodbolt) [inline static/settings.ts:416]: :+1:

File: static/themes.ts [hunk 2]
```diff
@@ -26,15 +26,15 @@ import $ from 'jquery';
 import {editor} from 'monaco-editor';
 import {SiteSettings} from './settings';
 
-export type Themes = 'default' | 'dark' | 'darkplus';
+export type Themes = 'default' | 'dark' | 'darkplus' | 'system';
```
Comment (mattgodbolt) [inline static/themes.ts:29]: I guess we're stuck with `default` being "light" now. We ca'nt rename that without some trickery, right? Mayybe worth making it "Light" and then having a one-time fixup when we load from the settings `if default return lght` stuff

**Review by mattgodbolt — APPROVED** [review]
Definitely up for a chat about default/light/system stuff

**Review by matsjla — APPROVED** [review]
Love the idea, great stuff

File: static/themes.ts (hunk 2 above)
Comment (matsjla) [reply static/themes.ts:29]: I think the key might be stuck to default, but we could use a different displayname

**Pull request merged**